### ISSUE TITLE
Cost of Optimization in Experiments

### ISSUE THREAD

**Issue by LeonResearch** [metauto-ai/GPTSwarm#28]
Hello GPTSwarm authors,

Thank you for such an interesting work on the multi-agent communication network! 

I am very interested in the idea of optimization of graph structure, and I have a question about the cost of the reinforcement optimization in your MMLU and Mini crosswords experiments.

In the paper, it seems like the optimization requires 200 iterations * 4 graph samples at each iteration for MMLU. This leads to a total of 800 queries of the multi-agent LLM system on the utility function, which is defined as the accuracy of 10% validation set (153 questions).

I haven't reproduced the experiment results yet, but it sounds like a big cost to get the optimized edge probability. Could you please share some information about the training cost during your experiments? 

Many thanks!
Leon

**Comment by mczhuge** [comment]
Hi Leo,

Here is the cost analysis. We will soon upload the newest version to Arxiv to supplement this. 

<img width="982" alt="image" src="https://github.com/user-attachments/assets/3e5e1a48-133e-41bf-85ec-24af99bd9318">

Our main focus is to propose a vision of how to build agents using graphs and related optimization methods. We agree with you that better optimization is highly valuable in this community.

Hope this could help you!

Mingchen
